dataset/reduction.cpp

```diff
@@ -59,9 +67,40 @@ DataArray mean(const DataArrayConstView &a, const Dim dim) {
       a, [](auto &&... _) { return mean(_...); }, dim, a.masks());
 }
 
+DataArray mean(const DataArrayConstView &a) {
+  auto count = isInt(a.data().dtype()) ? sum(isfinite(astype(a, dtype<double>)))
+                                       : sum(isfinite(a));
+  return sum(a) * (1.0 * units::one) / count;
```

Comment:
- Is the multiplication with `1.0` to convert to `double`? Shouldn't we keep `float` if input is float?
- Maybe also add parentheses so we do fewer divisions, or use `reciprocal(count)`?
- Would it be easier to support `int` in `isfinite` and friends instead of adding hacks here? `np.isfinite` also works with ints.

Reply:
> Is the multiplication with 1.0 to convert to double? Shouldn't we keep float if input is float?
Yes we should preserve floats. Use of *= rather than * should do that.
> Maybe also add parentheses so we do fewer divisions, or use reciprocal(count)
Yes 
> Would it be easier to support int in isfinite and friends instead of adding hacks here? np.isfinite also works with ints.
We explicitly do not support these checks with ints since there is not cross-platform support for this at the `std` level. Though we could move this behaviour into `isfinite`

Reply:
> We explicitly do not support these checks with ints since there is not cross-platform support for this at the std level. Though we could move this behaviour into isfinite

I think we probably should? Would make things easier here, and does actually make sense conceptually.

Reply:
Agreed. I have some some changes that do this (not pushed yet). I'm applying the conversion at the element level where needed (`if constexpr`)